simulator/modules/branch/branch.cpp

```diff
@@ -49,10 +49,13 @@ void Branch<FuncInstr>::clock( Cycle cycle)
      
     bool is_misprediction = false;
 
-    if ( instr.get_bp_data().is_taken != instr.is_taken())
-        is_misprediction = true;
-    else if ( instr.is_taken())
-        is_misprediction = instr.get_bp_data().target != instr.get_new_PC();
+    if ( instr.is_branch() || instr.is_indirect_jump())
```

Comment:
I suggest to rewrite it as:

```c++
if ( instr.is_branch())
    is_misprediction = instr.get_bp_data().is_taken != instr.is_taken();
else if ( instr.is_indirect_jump() || instr.is_direct_jump())
    is_misprediction = instr.get_bp_data().target != instr.get_new_PC();
```

Reply:
prediction of target in case of direct_jump is checked at decode stage, we should check it only once

Reply:
Yes, unless it is not disabled by a command line option (see below)

Reply:
But if we enable it - we can do flush twice: 
at decode stage and at memory stage.
It is wrong

Reply:
Because if target is predicted wrong, ```is_misprediction``` will be set as ```true``` both in decode module and in branch module

Reply:
OK, let's leave it as is (but slight clean up of `if's` would be nice).